tests/integration/test_remoted/test_agent_communication/test_invalid_protocols_communication.py

```diff
@@ -0,0 +1,150 @@
+receiver_sockets, monitored_sockets = None, None  # Set in the fixtures
+
+
+
+# Test function.
+@pytest.mark.parametrize('test_configuration, test_metadata',  zip(test_configuration, test_metadata), ids=cases_ids)
+def test_invalid_protocols_communication(test_configuration, test_metadata, configure_local_internal_options, truncate_monitored_files,
+                            set_wazuh_configuration, configure_sockets_environment_module, daemons_handler, simulate_agents,
+                       connect_to_sockets_module, waiting_for_analysisd_startup):
+
+    '''
+    description: Check agent-manager communication with several agents simultaneously via TCP, UDP or both.
+                 For this purpose, the test will create all the agents and select the protocol using Round-Robin. Then,
+                 an event and a message will be created for each agent created. Finally, it will search for
+                 those events within the messages sent to the manager.
+
+
+    parameters:
+        - test_configuration
+            type: dict
+            brief: Configuration applied to ossec.conf.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Configure the Wazuh local internal options using the values from `local_internal_options`.
+        - daemons_handler:
+            type: fixture
+            brief: Restart service once the test finishes stops the daemons.
+        - simulate_agents
+            type: fixture
+            brief: create agents
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+        - configure_sockets_environment_module:
+            type: fixture
+            brief: Configure environment for sockets and MITM.
+        - connect_to_sockets_module:
+            type: fixture
+            brief: Connect to a given list of sockets.
+        - waiting_for_analysisd_startup:
+            type: fixture
+            brief: Wait until the 'wazuh-analysisd' has begun and the 'alerts.json' file is created.
+    '''
+
+    agent = simulate_agents[0]
+
+
+    def validate_agent_manager_protocol_communication(protocol='TCP', manager_port=1514):
+
+
+        def send_event(event, protocol, manager_port, agent):
+            """Send an event to the manager"""
+
+            sender, injector = connect(agent, manager_port = manager_port, protocol = protocol, wait_status= '' if protocol == 'UDP' else 'active' )
+            sender.send_event(event)
+            injectors.append(injector)
+            return injector
+
+
+        # Generate custom events for each agent
+        search_pattern = f"test message from agent {agent.id}"
```

Comment:
Can these strings be defines?

Reply:
It is not necessary because it is part of the test.